src/menu/tray.js

```diff
@@ -83,17 +84,65 @@ export class Tray extends EventEmitter {
     if (!kubeconfigPath) {
       throw new Error('No kubeconfig path found');
     }
-    fs.watch(kubeconfigPath, () => {
-      this.updateContexts();
-      const contextMenu = electron.Menu.buildFromTemplate(this.#contextMenuItems);
-
-      this.#trayMenu.setContextMenu(contextMenu);
+    // Add a delay before starting the file system watcher to avoid hitting a lock on the file
+    // that is taking a while to be cleared.
+    // setTimeout(() => {
+    // Maybe the file exists now...
+    this.buildFromConfig(kubeconfigPath);
+    const watcher = fs.watch(kubeconfigPath);
+
+    watcher.on('error', (code, signal) => {
+      console.log(`Failed to fs.watch ${ kubeconfigPath }: code: ${ code }, signal: ${ signal }`);
     });
+    watcher.on('change', (eventType, _) => {
+      if (eventType === 'rename' && !kubeconfig.hasAccess(kubeconfigPath)) {
+        // File doesn't exist. Wait for it to be recreated
+        return;
+      }
+      this.buildFromConfig(kubeconfigPath);
+    });
+    // }, 1000);
```

Comment:
This is the other side of the not in use setTimeout. Removed?

Reply:
now it is